Functions/ImageProcessing/xASL_im_CreateASLDeformationField.m

```diff
@@ -62,7 +62,8 @@ function xASL_im_CreateASLDeformationField(x, bOverwrite, EstimatedResolution, P
 sKernel(EstimatedResolution<TemplateResolution) = 0;
 
 %% 2) Fill NaNs at edges y_T1.nii flowfield to prevent interpolation artifact
-xASL_im_FillNaNs(x.P.Path_y_T1, 3, x.settings.Quality); % First fill NaNs, to prevent interpolation artifacts
+xASL_im_FillNaNs(x.P.Path_y_T1, 3, [], [], true); % First fill NaNs, to prevent interpolation artifacts
```

Comment:
There are no zeroes filled here?

Anyway, in general: sometimes NaNs at the flowfield edges have turned into zeroes. So we want to convert zeroes to NaNs. Then we run our filling, which only fills stuff at the edges of the flowfield. Any NaNs (or original zeroes) that remain are in the middle of the flowfield, would be filled by the ndnan_filter. I agree with you in theory that this could go wrong, but in reality I don't think there will ever be a 0.0000000000.

Reply:
I agree. Don't close yet, I'll add details to comments and see if I can improve this a bit. But agree in general that this is needed.